adi/adpd410x.py

```diff
@@ -0,0 +1,105 @@
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class adpd410x(rx, context_manager):
+    """ adpd410x Multimodal Sensor Front End """
+
+    _complex_data = False
+    channel = []  # type: ignore
+    _rx_channel_names = [
+        "channel0",
+        "channel1",
+        "channel2",
+        "channel3",
+        "channel4",
+        "channel5",
+        "channel6",
+        "channel7",
+    ]
+    _device_name = ""
+
+    def __init__(self, uri=""):
+
+        context_manager.__init__(self, uri, self._device_name)
+
+        self._rxadc = self._ctx.find_device("adpd410x")
+
+        for name in self._rx_channel_names:
+            self.channel.append(self._channel(self._rxadc, name))
+
+    @property
+    def sampling_frequency(self):
+        """Get sampling frequency of the adpd410x"""
+        return None
+
+    @sampling_frequency.setter
+    def sampling_frequency(self, value):
+        """Sets sampling frequency of the adpd410x"""
+        self._set_iio_dev_attr_str("sampling_frequency", value, self._rxadc)
+
+    @property
+    def last_timeslot(self):
+        """Get last timeslot of the adpd410x"""
+        return None
+
+    @last_timeslot.setter
+    def last_timeslot(self, value):
+        """Sets last timeslot of the adpd410x"""
+        self._set_iio_dev_attr_str("last_timeslot", value, self._rxadc)
+
+    @property
+    def operation_mode(self):
+        """Get operation mode of the adpd410x"""
+        return None
```

Comment:
Can this be read back?

Reply:
Currently the driver is populated with setters only:
https://github.com/analogdevicesinc/no-OS/blob/5037d892196ec5c692452a624cc11d1b2561f724/drivers/photo-electronic/adpd410x/adpd410x.h#L1214-L1223

I believed the implementation of getters/adding device parameters should start from the no-OS driver and then added into the upper layers (iio no-OS driver, pyadi-iio support).

If you think it is necessary to have the readbacks, I can do the updates in the files mentioned above.

Reply:
Does that mean the reads will fail or they are meaningless?

Reply:
They are meaningless at this point in time. (Just added them as placeholders for future implementation).
I made some updates on the no-OS driver so I can add the read backs here also. Will update soon.